### ISSUE TITLE
Twig Runtime Error when evaluating key values of undefined variables

### ISSUE THREAD

**Issue by missmatsuko** [nystudio107/craft-emptycoalesce#3]
This is an example from the README. 
`woof` is undefined and 'bark' is printed (as expected).
```
{% set bar = null %}
{% set foo = '' %}
{% set baz = [] %}
{% set foobar = woof ??? bar ??? foo ??? baz ??? 'bark' %}
{{ foobar }} 
{# 'bark' #}
```

Here, `dog` is undefined, but I try to get `dog.sound`. 
I still expected 'bark' to be printed, but instead I get a Twig Runtime Error.
```
{% set bar = null %}
{% set foo = '' %}
{% set baz = [] %}
{% set foobar = dog.sound ??? bar ??? foo ??? baz ??? 'bark' %}
{{ foobar }} 
{#
  Twig Runtime Error – Twig_Error_Runtime
  Variable "dog" does not exist.
 #}
```

Is there any way around this error? 
If not, I think I'd still need to use longer checks like this, which I was hoping to avoid by using this plugin.
```
{% set bar = null %}
{% set foo = '' %}
{% set baz = [] %}
{% set foobar = dog is defined and dog.sound and dog.sound|length ? dog.sound : bar ??? foo ??? baz ??? 'bark' %}
{{ foobar }}
{# 'bark' #}
```

**Comment by joshuabaker** [comment]
I’m encountering this too.

```twig
{{ invalid.nested ?? 'valid' }}
{# string(5) "valid" #}

{{ invalid.nested ??? 'valid' }}
{#
  Twig Runtime Error – Twig_Error_Runtime
  Variable "invalid" does not exist.
#}
```

A workaround is to ‘simplify’ the variable using the native null coalescing operator first.

```twig
{% set simple = invalid.nested ?? null %}
{{ simple ??? 'valid' }}
{# string(5) "valid" #}
```

**Comment by missmatsuko** [comment]
Interesting, I hadn't tried the native null coalescing operator. The fact that it works better in this situation makes it seem like this issue might be a bug.

**Comment by khalwat** [comment]
hrm. Will have a look!

**Comment by khalwat** [comment]
So this is something that used to work properly, but was broken in the fix made here https://github.com/nystudio107/craft-emptycoalesce/issues/2

**Comment by khalwat** [comment]
Well, that was a fun journey down the Twig Node Parser rabbit hole. This is fixed in 1.0.3: https://github.com/nystudio107/craft-emptycoalesce/releases/tag/1.0.3

It passes both of these cases as expected:

```
    {% set bar = null %}
    {% set foo = '' %}
    {% set baz = [] %}
    {% set foobar = dog.sound ??? bar ??? foo ??? baz ??? 'bark' %}
    {{ foobar }}

    {% set foobar = craft.app.request.getQueryParam('empty_param')  ??? 'expected' %}
    {{ foobar }}
```
